dvc/tree/azure.py

```diff
@@ -36,9 +37,62 @@ def __init__(self, repo, config):
             container = self._az_config.get("storage", "container_name", None)
             self.path_info = self.PATH_CLS(f"azure://{container}")
 
-        self._conn_str = config.get(
-            "connection_string"
-        ) or self._az_config.get("storage", "connection_string", None)
+        self._get_string_token_credential(config)
+
+        if not (self._conn_str or self._credential):
+            from azure.identity import (
+                AzureCliCredential,
+                ChainedTokenCredential,
+            )
+
+            account_name = config.get("account_name")
+            self._account_url = f"https://{account_name}.blob.core.windows.net"
+            if not account_name:
+                logging.warning(
+                    "You must enter an account name for this auth flow"
+                )
+
+            # Microsoft azure docs
+            # https://docs.microsoft.com/en-us/python/api/overview/azure/identity-readme?view=azure-python
+            if config.get("azcli_credential"):
+                self._credential = ChainedTokenCredential(AzureCliCredential())
+            else:
+                self._credential = self._create_client_secret_cred(config)
+
+    def _create_client_secret_cred(self, config):
+        from azure.identity import ClientSecretCredential
+
+        client_id = config.get("client_id") or self._az_config.get(
+            "client", "id", None
+        )
+
+        client_secret = config.get("client_secret") or self._az_config.get(
+            "client", "secret", None
+        )
+
+        tenant_id = config.get("tenant_id") or self._az_config.get(
+            "tenant", "id", None
+        )
+
+        if not (client_id and client_secret and tenant_id):
+            logging.error(
+                "Not enough information to authenticate to azure remote."
+            )
+            return None
+
+        return ClientSecretCredential(
+            client_id=client_id,
+            client_secret=client_secret,
+            tenant_id=tenant_id,
+        )
+
+    def _get_string_token_credential(self, config):
+        # Get `token` like credential to pass to
+        # blob service client. String like
+        # conn_str, sas_token, account_key
+        self._conn_str = config.get("connection_string") or self._az_config.get(
+            "storage", "connection_string", None
+        )
```

Comment:
Not worth a separate method, just inline it

Reply:
The `__init__` method is getting way too crowded, it was a little hard to read. That's why I decided to make a the method `_get_string_token_credential` to hold this bit of logic.